bundles/org.openhab.binding.fritzboxtr064/src/main/java/org/openhab/binding/fritzboxtr064/internal/SOAPValueConverter.java

```diff
@@ -129,7 +129,7 @@ public SOAPValueConverter(HttpClient httpClient) {
                         return new DecimalType(rawValue);
                     }
                 default:
-                    return null;
+                    throw new IllegalStateException("Unrecognized data type: " + dataType);
```

Comment:
I disagree here. Returning null is valid and results in an empty optional. It's just our null annotations that can't detect that properly.

Reply:
Hm, any idea how to make it compile in the IDE then? The [suppress](https://github.com/J-N-K/openhab-addons/pull/9/files#diff-afc400f7492e2300158432d61dcd3f00L111) above does not work for me.